### DISCUSSION TITLE
What happens when n_lags = 0?

### DISCUSSION THREAD

**Discussion by thanasisKirmizis** [ourownstory/neural_prophet#243] [Q&A - get help using NeuralProphet] [answered]
Hello! So I am trying to wrap my head around that. You mention that if n_lags = 0, then auto-regression is not used. In that case, how exactly is the time-series forecasting done? Also, how was the forecasting done in the original Prophet, where n_lags was not an option? Did it not use auto-regression at all? I'm a bit confused! Thanks :)

**Comment by HansikaPH** [comment, marked as answer]
If n_lags=0, then the AR-Net is not included in the model (auto-regression not included). This means that the model functionality is similar to that in the original Prophet where the basic model is a regression model that incorporates additive components, such as trend, seasonality, events. In that mode, the autocorrelation of the time series is not captured and the regressors all have one-to-one mapping at every time step considered.

**Reply by ourownstory** [reply]
@thanasisKirmizis Adding to Hansika's great answer, regarding original Prophet: You assumed right, it did not implement auto-regression, or any other lagged regression.

**Reply by M-Adrian-N** [reply]
And then when we use n_lags=3 for e.g. will it be something like aY(t-1)+bY(t-2)+cY(t-3) or just cY(t-3) as an regressor in the equation?
Also, when using any n_lag>0 why it the forecast (yhat) not showing anymore?

Thank you

**Reply by ourownstory** [reply]
Hi @M-Adrian-N 

 > when we use n_lags=3 for e.g. will it be something like aY(t-1)+bY(t-2)+cY(t-3) 

Yes

> Also, when using any n_lag>0 why it the forecast (yhat) not showing anymore?

There should be `yhat1`, ... up to `n_forecasts`

**Reply by M-Adrian-N** [reply]
ah, so all forecasts don't stack up under yhat, but there is one forecasted point under every n-th yhat. So the interpretation for Yhat4=500 (assuming n_forecast=4) is then (assuming we are forecasting number of bike sold every day): the forecast for the 4th coming day is 500 bikes, considering the 4 steps/lags behind as regressors. Correct me pls if I'm wrong. thank you

**Reply by ourownstory** [reply]
@M-Adrian-N I think you got it almost right. The lags are independent of n_forecasts - it just is the number of inputs to the model.
yhat4 is in your case: the number of bikes forecasted to be sold today, as predicted 4 days ago (4 days old prediction).

**Comment by steplp** [comment]
I have encounter the same problems, when using any n_lag>0 why it the forecast (yhat) not showing anymore?

**Comment by steplp** [comment]
but we need forecast the future datas not the original datas
